File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        result = []
        left = 0
        right = len(numbers) - 1

        while left < right:
            if numbers[left] + numbers[right] == target:
                result.append(left + 1)
                result.append(right + 1)
                return result
            elif numbers[left] + numbers[right] > target:
                right -= 1
            else:
                left += 1
            

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Map to check again - uses numbers in array as the key and the index of those numbers are the values
        Map<Integer, Integer> prev = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            // Calculate the Difference
            int diff = target - nums[i];
            if (prev.containsKey(diff)){
                return new int[]{prev.get(diff), i};
            }
            prev.put(nums[i], i);
        }
        return null;
    }
}
